### DISCUSSION TITLE
Drivetrain Motor Controller (STAGE 2)

### DISCUSSION THREAD

**Discussion by conner-replogle** [Comet-Robotics/URC#19] [Drivetrain]
## Creating a Ros2 2 Node for the drivetrain
Now that we have a moving bot we need to be able to control it from many different parts of our rover. This means we need to build a ROS 2 Node with the whole purpose of driving it. 

### Setting up your enviornmnet
You will need ubuntu 22.04
[Window VM Setup](https://automaticaddison.com/how-to-install-ubuntu-20-04-and-virtualbox-on-your-computer/)
[Mac VM Setup](https://docs.getutm.app/guides/ubuntu/)

[Ros2 Node Tutorial](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html)

**Comment by conner-replogle** [comment]
[We need an action server](https://docs.ros.org/en/foxy/Tutorials/Intermediate/Writing-an-Action-Server-Client/Py.html)

**Comment by JonnyJonny2000** [comment]
IMU with GPS
- https://www.amazon.com/Navigation-Positioning-Microcontroller-Compatible-Sensitivity/dp/B0B31NRSD2/ref=asc_df_B0B31NRSD2/?tag=hyprod-20&linkCode=df0&hvadid=693308325730&hvpos=&hvnetw=g&hvrand=7725120964331958925&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9026839&hvtargid=pla-1822029456211&psc=1&mcid=842e9881bf6d32c28b3e3a77c36684b9
